### PR TITLE
RadioButton/Checkbox: color updates

### PR DESCRIPTION
**Pull request by mziemer21**
## TODO

- [ ] Accessibility checkup
- [x] Update documentation
- [x] Add/update Tests
- [ ] Pinterest Designer (for design changes)
- [x] [Format PR title](https://github.com/pinterest/gestalt/#releasing): `ComponentName: Description`.

### CODE REVIEW

**Comment by mziemer21** [comment]
This fixes https://github.com/pinterest/gestalt/issues/999 and https://github.com/pinterest/gestalt/issues/994

**Comment by mziemer21** [comment]
`borderColorLightGray` is also used by Box, and appears in a Table snapshot.

Screen Shot 2020-07-08 at 8 40 16 PM

File: packages/gestalt/src/Borders.css [hunk 1]
```diff
@@ -30,7 +30,7 @@
 }
 
 .borderColorLightGray {
-  border-color: var(--gestalt-border-color-lightGray);
+  border-color: var(--gestalt-colorGray100);
```
Comment (jennyscript) [inline packages/gestalt/src/Borders.css:33]: @ponori Are you ok with us changing most of the borders from `#ddd` to `#efefef` in light mode?

Screen Shot 2020-07-09 at 8 14 51 AM

I think we need a new color variable for the borders we can override in dark mode

Comment (AlbertCarreras) [reply packages/gestalt/src/Borders.css:33]: Resolved in  https://github.com/pinterest/gestalt/pull/1034

**Comment by jennyscript** [comment]
@mziemer21 I think we can close this one in favor of #1034

**Pull request closed without merging**